modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingResultTransformer.java

```diff
@@ -38,10 +42,15 @@ public Object applyChecked(ExecutionContextAdapter operationContext, Object valu
                                                       config,
                                                       operationContext,
                                                       connectionSupplier,
-                                                      supportsOAuth);
+                                                      supportsOAuth,
+                                                      getConnectionSpanInfo);
 
     ListConsumer<?> consumer = new ListConsumer(producer);
     consumer.loadNextPage();
     return new ConsumerStreamingIterator<>(consumer);
   }
+
+  public void setGetConnectionSpanInfo(InitialSpanInfo getConnectionSpanInfo) {
+    this.getConnectionSpanInfo = getConnectionSpanInfo;
+  }
```

Comment:
Not a fan of having this be mutable and optional. But I have no better proposal that doesn't involve changing too many things.

Reply:
I will leave this one for a future refactor.